Liner Regression Lab
Recently, the annual number of driver deaths per 100,000 for the selected age groups was as follows:


Age	Number of Driver Deaths per 100,000

    17.5	38

    22	    36

    29.5	24

    44.5	20

    64.5	18

    80	    28

Questions

For each age group, pick the midpoint of the interval for the x value. (For the 75+ group, use 80.)

Using “ages” as the independent variable and “Number of driver deaths per 100,000” as the dependent variable, make a scatter plot of the data.

Calculate the least squares (best–fit) line. Put the equation in the form of: ŷ = a + bx

Predict the number of deaths for ages 40 and 60.

Based on the given data, is there a linear relationship between age of a driver and driver fatality rate?

Find the correlation coefficient (Pearson's r). Is it significant?


In [37]:
from sklearn.linear_model import LinearRegression
# Ages as independent variable
X = [[19.75],[25.75],[37],[54.5],[72.25],[80]]
# Number of driver deaths dependent variable
y = [38,36,24,20,18,28]
reg = LinearRegression().fit(X, y)

In [38]:
print(reg.score(X, y))
print(reg.coef_)
print(reg.intercept_)

0.481572239598
[-0.23146209]
38.4917347095


In [39]:
print(reg.predict(50))

[ 26.91863043]


In [40]:
from scipy import stats
stats.pearsonr([e[0] for e in X], y)

(-0.69395406159079709, 0.12616341342313486)

Yes, the correlation coefficient is significant. According to the documentation:

Like other correlation coefficients, this one varies between -1 and +1 with 0 implying no correlation.
Correlations of -1 or +1 imply an exact linear relationship.
Positive correlations imply that as x increases, so does y.

The correlation coefficient is -0.69, which means as age increases the number of accidnets drop.